File: Assignment_5/Assignment5_1.py
"""-------------------------------------------------------------------------------------------
                          Assignment 5_1 
                    (Student name - Vaishali Jorwekar)
                     Python by Marvellous Infosystems
---------------------------------------------------------------------------------------------
Problem statement : This program accepts two numeric inputs from the user and performs 
                    artimatic operations on them. 
-----------------------------------------------------------------------------------------"""
#-----------------------------------------------------------------------------------
#This function accepts two input parameters and performs Addition,Subtraction, 
# Multiplication and Division operations on them
#-----------------------------------------------------------------------------------
# Lambda functions
#Addition function 
Addition = lambda num1,num2:num1+num2
#Subtraction function
Subtraction=lambda num1,num2:num1-num2
#Multiplication
Product=lambda num1,num2:num1*num2
#Division
Division=lambda num1,num2:num1/num2
#-----------------------------------------------------------------------------------

def performOperations(num1,num2,choice):
    print("------------------------------------------------------")
    if choice==1:
        print(f"Sum: {num1} + {num2} =",Addition(num1,num2))
    elif choice==2:    
        print(f"Difference: {num1} - {num2} =",Subtraction(num1,num2))
    elif choice==3:    
        print(f"Product: {num1} * {num2} =",Product(num1,num2))
    elif choice==4:    
        if (num2 != 0 ) : 
          print(f"\nDivision: {num1} / {num2} = ",Division(num1,num2))
        else :
            print("\nDivision is not possible")    
    print("------------------------------------------------------")

#-----------------------------------------------------------------------------------
# This function accepts two input numbers from users and calls function 
# performOperations with these two input parameters
#-----------------------------------------------------------------------------------

def main():
    number1=int(input("Enter first number:"))
    number2=int(input("Enter second number:"))
    choice = 0
    while not (choice == 5):
        print(f"\nPerforming following arithmatic operations on {number1} and {number2}:\n" \
        "1. Addition\n" \
        "2. Subtraction\n"\
        "3. Multiplication\n" \
        "4. Division\n"
        "5. Exit")
        choice = int(input("Enter operation number(1/2/3/4/5):"))
        performOperations(number1,number2,choice)
#-----------------------------------------------------------------------------------

#-----------------------------------------------------------------------------------
# This is main entry point of the program
#-----------------------------------------------------------------------------------
if __name__=="__main__":
    main()
#-----------------------------------------------------------------------------------

File: JavaPractice/Assignment_5/Assignment5_1.java
import java.util.Scanner;
/*-------------------------------------------------------------------------------------------
                          Assignment 5_1 
                    (Student name - Vaishali Jorwekar)
                     Python by Marvellous Infosystems
---------------------------------------------------------------------------------------------
Problem statement : This program accepts two numeric inputs from the user and performs 
                    arithmetic operations on them. 
-----------------------------------------------------------------------------------------"*/

public class Assignment5_1 {

	public static void main(String[] args) {
		Scanner inputObj = new Scanner(System.in);
		System.out.print("Enter First Number:"); 
		int number1 = inputObj.nextInt();
		System.out.print("Enter Second Number:"); 
		int number2 = inputObj.nextInt();
		processNumbers(number1,number2, inputObj);
		inputObj.close();

	}

	private static void processNumbers(int number1, int number2,Scanner inputObj) {
		// TODO Auto-generated method stub
		int choice=0;
		String continueProcess= "yes";
		while(continueProcess.equalsIgnoreCase("yes")|| continueProcess.equalsIgnoreCase("y")) {
			System.out.println("\n1.Addition\n2.Subraction\n3.Multiplication\n4.Division\n5.Exit");
			System.out.println("\nEnter the choice of operation:");

			choice=inputObj.nextInt();
			switch(choice) {
			case 1:
				ArtithmeticModule add=(num1,num2)->num1+num2;
				int result=add.Arithmetic(number1,number2);
				System.out.println("Addition of "+number1+"+"+number2+" is :"+result);
				break;




			case 2:
				ArtithmeticModule sub=  (num1,num2) -> num1-num2;
				result=sub.Arithmetic(number1,number2);
				System.out.println("Subtraction of "+number1+"-"+number2+" is :"+result);
				break;


			case 3:
				ArtithmeticModule multi=  (num1,num2) -> num1*num2;
				result=multi.Arithmetic(number1,number2);
				System.out.println("Multiplication of "+number1+"*"+number2+" is :"+result);
				break;

			case 4:
				if(number2==0) {
					System.out.println("Division not possible...");
				}else {
					ArtithmeticModule div=  (num1,num2) -> num1/num2;
					result=div.Arithmetic(number1,number2);
					System.out.println("Division of "+number1+"*"+number2+" is :"+result);
				}
				break;

			default:
				System.out.println("Invalid choice....");

			}

			System.out.println("Do you want to contuinue(yes/no)?:");
			continueProcess= inputObj.next();
		}
	}

}
